src/Tribe/Plugins_API.php

```diff
@@ -0,0 +1,310 @@
+		$result_slugs = wp_list_pluck( $result->plugins, 'slug' );
+
+		// get all of our products
+		$products = $this->get_products();
+
+		// we don't ever want to show event-aggregator as a plugin
+		unset( $products['event-aggregator'] );
+
+		$count = 0;
+		$products_to_inject = array();
+		foreach ( $products as $key => $product ) {
+			$products[ $key ] = $product = $this->build_product_data( $product );
+
+			// if the product is already installed, skip it
+			if ( $product['is_installed'] ) {
+				continue;
+			}
+
+			// if the product is already in the results, skip it
+			if ( in_array( $product['slug'], $result_slugs ) ) {
+				continue;
+			}
+
+			$products_to_inject[] = $product;
+
+			$count++;
+
+			if ( 3 === $count ) {
+				break;
+			}
+		}
+
+		// prepend the products that we wish to inject
+		for ( $i = count( $products_to_inject ) - 1; 0 <= $i; $i-- ) {
+			array_unshift( $result->plugins, $products_to_inject[ $i ] );
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Convert our own product data to match API results
+	 *
+	 * @since tbd
+	 *
+	 * @param array $product_data
+	 *
+	 * @return array
+	 */
+	public function build_product_data( $product_data ) {
+		$defaults = array(
+			'name' => null,
+			'slug' => null,
+			'version' => null,
+			'author' => '<a href="https://theeventscalendar.com">Modern Tribe, Inc.</a>',
+			'author_profile' => null,
+			'requires' => '3.9',
+			'tested' => '4.7.5',
+			'rating' => null,
+			'ratings' => array(),
+			'num_ratings' => null,
+			'support_threads' => null,
+			'support_threads_resolved' => null,
+			'active_installs' => null,
+			'downloaded' => null,
+			'last_updated' => null,
+			'added' => null,
+			'homepage' => '',
+			'sections' => array(),
+			'short_description' => null,
+			'download_link' => '',
+			'screenshots' => array(),
+			'tags' => array(),
+			'versions' => array(),
+			'donate_link' => null,
+			'icons' => array(
+				'default' => null,
+			),
+			'contributors' => array(),
+			'tribe-result' => true,
+		);
+
+		$product = array_merge( $defaults, $product_data );
+
+		if ( ! empty( $product['title'] ) && empty( $product['name'] ) ) {
+			$product['name'] = $product['title'];
+		}
+
+		if ( ! empty( $product['description'] ) && empty( $product['short_description'] ) ) {
+			$product['short_description'] = wp_trim_words( $product['description'], 27 );
+		}
+
+		if ( ! empty( $product['image'] ) && empty( $product['icons']['default'] ) ) {
+			if ( 0 === strpos( $product['image'], 'http' ) ) {
+				$product['icons']['default'] = $product['image'];
+			} else {
+				$product['icons']['default'] = Tribe__Main::instance()->plugin_url . '/src/resources/' . $product['image'];
+			}
+		}
+
+		return $product;
+	}
+
+	/**
+	 * Get product info
+	 *
+	 * @since tbd
+	 *
+	 * @return array
+	 */
+	public function get_products() {
+		$products = array(
+			'the-events-calendar' => array(
+				'title' => __( 'The Events Calendar', 'tribe-common' ),
```

Comment:
That's overkill here. It should be escaped on output. The above `esc_html()` for the `filter_action_links()` method are being done because that is the latest influence we have over the output of the links and escaping at that point makes sense.